src/ApacheOrcDotNet.OptimizedReader/OrcContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using ApacheOrcDotNet.Protocol;
+
+namespace ApacheOrcDotNet.OptimizedReader
+{
+    public class OrcContext
+    {
+        public OrcContext(CompressionKind compressionKind, int compressionBlockSize, int maxValuesToRead, int maxCompressedBufferLength = 0, int maxDecompresseBufferLength = 0)
+        {
+            CompressionKind = compressionKind;
+            CompressionBlockSize = compressionBlockSize;
+            MaxValuesToRead = maxValuesToRead;
+
+            if (maxCompressedBufferLength <= 0)
+                maxCompressedBufferLength = 25 * 1024 * 1024;
```

Comment:
On earlier versions, we were loading more data than needed and sometimes we would request around 7Mb of data to be decompressed. That data was loaded into buffers with the maximum size of `maxCompressedBufferLength`.

The above is no longer true, though. After the changes that calculate the data ranges based on the positions, we may need to load a maximum of two compressed blocks, determined by the `compressionBlockSize` length. I updated the code to reflect that.

On the other hand, we don't know the size of the decompressed data until we actually decompress it. To account for that, based on the information from previous tests (when we were loading unnecessary extra data), the maximum decompressed length that we reach, using production data, was around~10Mb and to be sure that we would not run out of space in some edge case, an arbitrary number of 25Mb was used to store decompressed data.

After the data ranges calculation, it is unlikely that we will exceed the size of that buffer (now that we decompress a maximum of two `compressionBlockSize` chunks at a time).

P.S.: The above was implemented to drop the need for sequences and sequence reader.

Reply:
I'm not quite tracking.

The ORC postscript provides us with the maximum buffer size that will ever be decompressed from a single chunk.  A compressed chunk will never be larger than a decompressed chunk.

When we talk about buffer sizes here are we talking about chunks or RowGroup-delineated stream data?  Assume that a RowGroup is 100,000 rows long rather than 10,000.  Now include 10 bytes in each row value.  The default chunk size is 256KB, so we'll need 100,000*10/256k=4 chunks of uncompressed data for each RowGroup.  When we're using maxDecompressedBufferLength do we need to have space for all 4 chunks, or do we only decompress a chunk at a time?

Our ORC files are built with some defaults but others may have drastically different defaults, so hardcoding a buffer size isn't very safe...